code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/projects/ProjectMemberResourceManager.java

```diff
@@ -55,14 +55,23 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
         Policy policy = (Policy) ApiContext.getContext().getPolicy();
         String id = RequestUtils.makeSingularStringIfCan(criteria.get("id"));
         if (StringUtils.isNotEmpty(id)) {
-            ProjectMember projectMember = authDao.getProjectMember(Long.valueOf(id));
+            ProjectMember projectMember;
+            try {
+                 projectMember = authDao.getProjectMember(Long.valueOf(id));
+            } catch (NumberFormatException e) {
+                return null;
```

Comment:
I don't get the inconsistency then. Why is it ok to raise the exception if projectMember is null but not ok to do so if the id is garbage? IMO, they should behave exactly the same.